Mathlib/Analysis/Calculus/Deriv/Shift.lean

```diff
@@ -0,0 +1,28 @@
+/-
+Copyright (c) 2023 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import Mathlib.Analysis.Calculus.Deriv.Add
+import Mathlib.Analysis.Calculus.Deriv.Comp
+
+/-!
+### Invariance of the derivative under translation
+
+We show that if a function `h` has derivative `h'` at a point `a + x`, then `h (a + ·)`
+has derivative `h'` at `x`. Similarly for `x + a`.
+-/
+
+/-- Translation in the domain does not change the derivative. -/
+lemma HasDerivAt.comp_const_add {𝕜 : Type*} [NontriviallyNormedField 𝕜] (a x : 𝕜) {𝕜' : Type*}
+    [NontriviallyNormedField 𝕜'] [NormedAlgebra 𝕜 𝕜'] {h : 𝕜 → 𝕜'} {h' : 𝕜'}
```

Comment:
You don't need the codomain to be a normed algebra. A normed space is enough.

Reply:
OK, thanks; changed.